tempesta_fw/cache.c

```diff
@@ -1212,7 +1209,7 @@ tfw_wq_tasklet(unsigned long data)
 
 	while (!tfw_wq_pop(&ct->wq, &cw)) {
 		if (cw.resp) {
-			cw.action(cw.req, cw.resp);
+			tfw_cache_add(cw.resp, cw.req, cw.action);
```

Comment:
Thanks for implementing this. `tfw_cache_add()` is called from the right place now.